packages/ra-core/src/core/useGetResourceLabel.ts

```diff
@@ -0,0 +1,44 @@
+import inflection from 'inflection';
+import { useSelector } from 'react-redux';
+import { getResources } from '../reducer';
+import { useTranslate } from '../i18n';
+
+/**
+ * A hook which will return a translated resource name. It will use the label option of the `Resource` component if provided.
+ *
+ * @param resource The resource name
+ * @param pluralize A boolean indicating whether the resource label should be pluralized.
+ * @param lowFirstLetter A boolean indicating whether the first letter of the resource label should be lower cased.
+ */
+export const useGetResourceLabel = (): GetResourceLabel => {
+    const resources = useSelector(getResources);
+    const translate = useTranslate();
+
+    return (resource: string, pluralize = true): string => {
```

Comment:
Why is it a higher-order function and why isn't the hook simply returning the label? This needs to be explained in the jsDoc I think.